### PR TITLE
*: Use snapshot interceptor to do temporary table's point_get/batch_point_get

### LINKED ISSUE #28142: Use snapshot interceptor to do temporary table's point_get/batch_point_get

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #28142

### What is changed and how it works?

Use snapshot interceptor to do temporary table's point_get/batch_point_get

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -3048,3 +3058,7 @@ func (s *session) updateTelemetryMetric(es *executor.ExecStmt) {
 func (s *session) GetBuiltinFunctionUsage() map[string]uint32 {
 	return s.builtinFunctionUsage
 }
+
+func (s *session) getSnapshotInterceptor() kv.SnapshotInterceptor {
+	return temptable.SessionSnapshotInterceptor(s)
```
Comment (djshow832) [inline session/session.go:3063]: Can you add a test case to verify that the snapshot interceptor exists?
For example, begin a transaction, do some batch get, and get the snapshot interceptor through the session.

**Review by djshow832 — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**